### PR TITLE
Reorganizing build.md

### PR DESCRIPTION
**Pull request by jkaeser**

### CODE REVIEW

File: docs/build.md [hunk 1]
```diff
@@ -21,17 +21,24 @@ assets:
...
 
-### Steps
+## Steps
 
 Steps are the commands to run for the build. These are typically steps for setting up a site and running tests. 
 
 Each step must have a `name`. The name of each step is the build context, and will get its own status updates.
 
+### Plugins
+
```
Comment (jodyHamilton) [inline docs/build.md]: Move the introduction about plugins here. "Each step must use a step plugin. The default plugin is shell... Each plugin takes different additional parameters...

File: docs/build.md [hunk 2]
```diff
@@ -41,7 +48,37 @@ steps:
...
+    command: 'gunzip -c $ASSET_DIR/dev.sql.gz | `mysql foo` ; rm $ASSET_DIR/dev.sql.gz'
+  - name: Move code in place
+    command: 'mv $SRC_DIR /var/www/foo/code'
+  - name: Run behat tests
+    command: 'cd /var/www/foo/code/tests ; composer install ; bin/behat'
+{% endhighlight %}
+
+###### Developing a Drupal module:
+
+{% highlight yaml%}
```
Comment (jodyHamilton) [inline docs/build.md]: Get rid of this example. it's deprecated by the new drupal plugin

File: docs/build.md [hunk 3]
```diff
@@ -101,31 +138,9 @@ Here is the list of available options:
   - Whether to revert features using `drush fra` after the build is finished.
   - Accepts a **boolean** value.
 
-#### Available variables:
```
Comment (jodyHamilton) [inline docs/build.md]: Be consistent with the use of terms parameter vs variable (they mean the same thing but stick with parameter in these docs)

Also be consistent in each plugin section on having a section for Parameters in each of them. (Add a parameters section for shell and document command parameter same way you document these)

Comment (jkaeser) [inline docs/build.md]: Don't these variables differ from the available parameters? To me the difference is that parameters are the items the user can set a value to, e.g. "command: xyz". The user doesn't set a value to the variables but rather calls them in a command, e.g. "command: 'mv $SRC_DIR /var/www/foo/code'"

Comment (jodyHamilton) [inline docs/build.md]: Yes, sorry, I didn't realize this was a separate section on the variables.

What we need is a 'Parameters' section under both shell and drupal plugin that show each available parameter. So under Steps is Plugins, under Plugins is each plugin, and under each plugin is Parameters

Comment (jkaeser) [inline docs/build.md]: What do you think about putting the parameters in tables? e.g.

| Directory Configuration |                                                                                                 |
|-------------------------|-------------------------------------------------------------------------------------------------|
|`makeFile`               | The name of the make file to run to generate the install directory. Accepts a **string** value. |
|`profileName`            | The profile name used in creating a symlink to this directory if a make file is specified with the                             `makeFile` option and used to select the profile to install if the `runInstall` option is selected.                              Accepts a **string** value.                                                                    |
|`runInstall`             | If set, run `drush site-install` to perform a fresh install of the site using the `profileName` as                             the install profile and allowing the `installArgs` option to configure the install. Accepts a                                  **boolean** value.                                                                              |

Could do it with HTML to get some colspan up in that header

**Pull request merged**